### ISSUE TITLE
Update scripts and YML to use "Agent.OS" instead of custom variables like "OSVmImage"

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-sdk-tools#5494]
I believe many of our scripts and YML use custom variables like `OSVmImage` to detect operating system.  However, I think it might be better to use the built-in variable `Agent.OS` instead.

![image](https://user-images.githubusercontent.com/9459391/219535137-b9350040-ba2e-45ce-9a93-1d3976c9e1a2.png)

https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#agent-variables-devops-services

**Comment by danieljurek** [comment]
Correct, we try to put the OS marketing version in the output: https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/scripts/Get-BinarySizes.ps1#L56
